koku/subs/subs_data_extractor.py

```diff
@@ -22,11 +22,89 @@
 from reporting.models import SubsIDMap
 from reporting.models import SubsLastProcessed
 from reporting.provider.aws.models import TRINO_LINE_ITEM_TABLE as AWS_TABLE
+from reporting.provider.azure.models import TRINO_LINE_ITEM_TABLE as AZURE_TABLE
+
 
 LOG = logging.getLogger(__name__)
 
 TABLE_MAP = {
     Provider.PROVIDER_AWS: AWS_TABLE,
+    Provider.PROVIDER_AZURE: AZURE_TABLE,
+}
+
+ID_COLUMN_MAP = {
+    Provider.PROVIDER_AWS: "lineitem_usageaccountid",
+    Provider.PROVIDER_AZURE: "COALESCE(subscriptionid, subscriptionguid)",
+}
+
+RECORD_FILTER_MAP = {
+    Provider.PROVIDER_AWS: (
+        " lineitem_productcode = 'AmazonEC2' AND lineitem_lineitemtype IN ('Usage', 'SavingsPlanCoveredUsage') "
+        "AND product_vcpu != '' AND strpos(lower(resourcetags), 'com_redhat_rhel') > 0"
+    ),
+    Provider.PROVIDER_AZURE: (
+        " metercategory = 'Virtual Machines' AND chargetype = 'Usage' "
+        "AND json_extract_scalar(lower(additionalinfo), '$.vcpus') IS NOT NULL "
+        "AND json_extract_scalar(lower(tags), '$.com_redhat_rhel') IS NOT NULL"
+    ),
+}
+
+RESOURCE_ID_FILTER_MAP = {
+    Provider.PROVIDER_AWS: (
+        " AND lineitem_productcode = 'AmazonEC2' "
+        "AND strpos(lower(resourcetags), 'com_redhat_rhel') > 0 AND lineitem_usageaccountid = {{usage_account}}"
+    ),
+    Provider.PROVIDER_AZURE: (
+        " AND metercategory = 'Virtual Machines' "
+        "AND json_extract_scalar(lower(additionalinfo), '$.vcpus') IS NOT NULL "
+        "AND json_extract_scalar(lower(tags), '$.com_redhat_rhel') IS NOT NULL "
+        "AND (subscriptionid = {{usage_account}} or subscriptionguid = {{usage_account}}) "
+    ),
+}
+
+RESOURCE_SELECT_MAP = {
+    Provider.PROVIDER_AWS: " SELECT lineitem_resourceid, max(lineitem_usagestartdate) ",
+    Provider.PROVIDER_AZURE: " SELECT coalesce(resourceid, instancename), max(coalesce(date, usagedatetime)) ",
```

Comment:
Same here:
```suggestion
    Provider.PROVIDER_AZURE: " SELECT coalesce(NULLIF(resourceid, ""), instancename), max(coalesce(date, usagedatetime)) ",
```
The coalesce on the dates is ok.